TileStache/Goodies/VecTiles/transform.py

```diff
@@ -1563,6 +1670,76 @@ def generate_label_features(
         return label_feature_layer
 
 
+def generate_address_points(
+        feature_layers, zoom, source_layer=None, start_zoom=0):
+    """
+    Generates address points from building polygons where there is an
+    addr:housenumber tag on the building. Removes those tags from the
+    building.
+    """
+
+    assert source_layer, 'generate_address_points: missing source_layer'
+
+    if zoom < start_zoom:
+        return None
+
+    layer = _find_layer(feature_layers, source_layer)
+    if layer is None:
+        return None
+
+    new_features = []
+    for feature in layer['features']:
+        shape, properties, fid = feature
+
+        addr_housenumber = properties.get('addr_housenumber')
+
+        # consider it an address if the name of the building
+        # is just a number.
+        name = properties.get('name')
+        if name is not None and re.match('^[0-9-]+$', name):
+            if addr_housenumber is None:
+                addr_housenumber = properties.pop('name')
+
+            # and also suppress the name if it's the same as
+            # the address.
+            elif name == addr_housenumber:
+                properties.pop('name')
+
+        # We only want to create address points for polygonal
+        # buildings with address tags.
+        if shape.geom_type not in ('Polygon', 'MultiPolygon') or \
+           addr_housenumber is None:
```

Comment:
This is prematurely optimizing, but we might want to consider doing the geom_type check first to avoid the regular expression check, which I think would be more costly than string compares. But then again, I think it's more clear the way it is now so maybe it's best to save these kinds of things for a general performance pass afterwards.